homeassistant/components/airzone/config_flow.py

```diff
@@ -42,14 +40,14 @@ async def async_step_user(
 
         if user_input is not None:
             system_id = user_input.get(CONF_ID, DEFAULT_SYSTEM_ID)
+            entry_data: dict[str, Any] = {
+                CONF_HOST: user_input[CONF_HOST],
+                CONF_PORT: user_input[CONF_PORT],
+            }
 
-            self._async_abort_entries_match(
-                {
-                    CONF_HOST: user_input[CONF_HOST],
-                    CONF_PORT: user_input[CONF_PORT],
-                    CONF_ID: system_id,
-                }
-            )
+            if system_id != DEFAULT_SYSTEM_ID:
+                entry_data[CONF_ID] = system_id
```

Comment:
This is a bit confusing to use a `.get` to resolve if the field is set.

I'd refactor this to instead check if CONF_ID is in user_input and set the values

Reply:
3df756e483c7e53a84a72ae65086eb26e959ba6a isn't exactly what you suggested, but it should be less confusing now.